### ISSUE TITLE
error when loading necromancers or vampires

### ISSUE THREAD

**Issue by masseymo** [robertjanetzko/LegendsBrowser#71]
I get the following when trying to open a page for dwarf necromancers or vampires.
Human necromancer pages do open.

org.apache.velocity.exception.MethodInvocationException: Invocation of method 'getShortDescription' in class legends.model.events.HfLearnsSecretEvent threw exception java.lang.ArrayIndexOutOfBoundsException: 1 at events.vm[line 29, column 40]

I just updated to DFhack 0.47.04-r2 Which is when I got the error in LB 1.18.2.
So I updated to 1.19.2 but the error persists.

**Comment by McArcady** [comment]
Did you export the legend information using command 'exportlegends' of dfhack-0.47.04-r2 ? or was it an old export ?
Could you attach the file region...-legends_plus.xml to this issue? 
Thanks!

**Comment by masseymo** [comment]
It is a fresh export from dfhack-0.47.04-r2.
Its a large region so the plus xml is over 100mb. 

I don't know if this info is helpful, but I did some extra searching:
vampire "Monom Laborowl"
legends browser is trying to access "hf/33337"
this unit is using a false identity 'Lòr Heatedlanced (Vúshkûbuk)' and that can be accessed at "/id/4087"

I thought it might be a problem with false identities but then "Rovod Chanttin (Onshenakrul)" is a necromancer and I can't access their page "/hf/5686"

[region1-00201-10-15-legends_plus.xml.zip](https://github.com/robertjanetzko/LegendsBrowser/files/5287155/region1-00201-10-15-legends_plus.xml.zip)

**Comment by McArcady** [comment]
No idea where the problem is... could you upload file region1-00201-10-15-legends.xml too?

**Comment by masseymo** [comment]
I can't upload here, over 10mb. I also don't seem to be able to post links..
its on google drive at the address below:
drive.google.com/file/d/1UO_yDfchneUSWodHQRx47RCrSbVWhzl1/view?usp=sharing

**Comment by McArcady** [comment]
Thanks! 
Using your export, I was able to access all vampires and necromancers.
Maybe some error occurred in your environment at loading. Do you have a log of LB starting and loading the export ?
If not, it is possible to generate one by starting LB from a terminal with this command:    
`java -jar path/to/lb/legendsbrowser-1.19.2-shaded.jar -w path/to/xml/region1-00201-10-15-legends.xml`
(replace the paths with the actual ones on your system)

**Comment by masseymo** [comment]
This ok?
[legendsbrowser.txt](https://github.com/robertjanetzko/LegendsBrowser/files/5344508/legendsbrowser.txt)

**Comment by McArcady** [comment]
Are you certain you have tried with LB v1.19.2 ?
All those "SEVERE" exceptions have been fixed in v1.19.2.
I was able to reproduce the bugs with your export using LB < v1.19.2.

**Comment by robertjanetzko** [comment]
yes, this log file is from an old version. did you launch the right version of legends browser after your update?
